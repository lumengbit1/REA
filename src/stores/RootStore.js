import { configure } from 'mobx';

import FormStore from './FormStore/FormStore';

configure({ enforceActions: 'always' });

class RootStore {
    constructor() {
        this.formStore = new FormStore(this);
    }
}

export default RootStore;
