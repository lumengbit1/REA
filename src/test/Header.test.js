import React from 'react';
import RootStore from '../stores/RootStore';
import { Provider } from 'mobx-react';
import Header from '../components/Header';
import { render, fireEvent } from 'react-testing-library';
// import { renderHook, act } from 'react-hooks-testing-library';
import sinon from 'sinon';

const rootStore = new RootStore();
describe('Render Test', function () {
    it('case: expect rendering correct', () => {
        const { container } = render(
            <Provider rootStore={rootStore}>
                <Header />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});

describe('Event Test', () => {
    it('case: expect Price Choose Click correct', () => {
        const updateProperty = sinon.stub(rootStore.formStore, 'updateProperty');
        const { getByTestId } = render(
            <Provider rootStore={rootStore}>
                <Header />
            </Provider>
        );

        fireEvent.click(getByTestId('choose'));

        expect(updateProperty.calledOnce).toBe(true);
        updateProperty.restore();
    });

    it('case: expect Reset Click correct', () => {
        const resetProperty = sinon.stub(rootStore.formStore, 'resetProperty');
        const { getByTestId } = render(
            <Provider rootStore={rootStore}>
                <Header />
            </Provider>
        );

        fireEvent.click(getByTestId('reset'));

        expect(resetProperty.calledOnce).toBe(true);
        expect(getByTestId('test').value).toBe('');
        resetProperty.restore();
    });

    it('case: expect onChange correct', () => {
        const change = sinon.stub(Header, 'getPrice');
        const { container, getByTestId } = render(
            <Provider rootStore={rootStore}>
                <Header />
            </Provider>
        );

        fireEvent.change(getByTestId('test'), { target: { value: 1 } });
        fireEvent.click(getByTestId('choose'));

        expect(change.calledOnce).toBe(true);
        change.restore();
    });
});
