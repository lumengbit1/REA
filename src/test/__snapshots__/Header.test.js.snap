// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test case: expect rendering correct 1`] = `
<div>
  <div
    class="header"
  >
    <label
      class="priceLabel"
      for="price"
    >
      Price:
    </label>
    <input
      data-testid="test"
      id="price"
      name="priceInput"
      type="text"
    />
    <button
      class="headerBtn"
      data-testid="choose"
    >
      Price Choose
    </button>
    <button
      class="headerBtn"
      data-testid="reset"
    >
      Reset
    </button>
  </div>
</div>
`;
